### YamlMime:TSType
name: DurableError
uid: durable-functions.DurableError
package: durable-functions
summary: >-
  A specfic error thrown when a scheduled activity or suborchestrator has
  failed.

  This error can be checked for via `instanceof` guards to catch only exceptions
  thrown

  by the DurableJS library.
fullName: DurableError
remarks: ''
isDeprecated: false
type: class
constructors:
  - name: DurableError(undefined | string)
    uid: durable-functions.DurableError.constructor
    package: durable-functions
    summary: Constructs a `DurableError` instance.
    remarks: ''
    isDeprecated: false
    syntax:
      content: 'new DurableError(message: undefined | string)'
      parameters:
        - id: message
          type: undefined | string
          description: error message.
inheritedProperties:
  - name: message
    uid: durable-functions.DurableError.message
    package: durable-functions
    summary: ''
    fullName: message
    remarks: ''
    isDeprecated: false
    syntax:
      content: 'message: string'
      return:
        description: ''
        type: string
    inheritanceDescription: <b>Inherited From</b> Error.message
  - name: name
    uid: durable-functions.DurableError.name
    package: durable-functions
    summary: ''
    fullName: name
    remarks: ''
    isDeprecated: false
    syntax:
      content: 'name: string'
      return:
        description: ''
        type: string
    inheritanceDescription: <b>Inherited From</b> Error.name
  - name: stack
    uid: durable-functions.DurableError.stack
    package: durable-functions
    summary: ''
    fullName: stack
    remarks: ''
    isDeprecated: false
    syntax:
      content: 'stack?: string'
      return:
        description: ''
        type: string
    inheritanceDescription: <b>Inherited From</b> Error.stack
  - name: stackTraceLimit
    uid: durable-functions.DurableError.stackTraceLimit
    package: durable-functions
    summary: >-
      The `Error.stackTraceLimit` property specifies the number of stack frames

      collected by a stack trace (whether generated by `new Error().stack` or

      `Error.captureStackTrace(obj)`).


      The default value is `10` but may be set to any valid JavaScript number.
      Changes

      will affect any stack trace captured _after_ the value has been changed.


      If set to a non-number value, or set to a negative number, stack traces
      will

      not capture any frames.
    fullName: stackTraceLimit
    remarks: ''
    isDeprecated: false
    syntax:
      content: 'static stackTraceLimit: number'
      return:
        description: ''
        type: number
    inheritanceDescription: <b>Inherited From</b> Error.stackTraceLimit
inheritedMethods:
  - name: captureStackTrace(object, Function)
    uid: durable-functions.DurableError.captureStackTrace
    package: durable-functions
    summary: >-
      Creates a `.stack` property on `targetObject`, which when accessed returns

      a string representing the location in the code at which

      `Error.captureStackTrace()` was called.


      ```js

      const myObject = {};

      Error.captureStackTrace(myObject);

      myObject.stack;  // Similar to `new Error().stack`

      ```


      The first line of the trace will be prefixed with

      `${myObject.name}: ${myObject.message}`.


      The optional `constructorOpt` argument accepts a function. If given, all
      frames

      above `constructorOpt`, including `constructorOpt`, will be omitted from
      the

      generated stack trace.


      The `constructorOpt` argument is useful for hiding implementation

      details of error generation from the user. For instance:


      ```js

      function a() {
        b();
      }


      function b() {
        c();
      }


      function c() {
        // Create an error without stack trace to avoid calculating the stack trace twice.
        const { stackTraceLimit } = Error;
        Error.stackTraceLimit = 0;
        const error = new Error();
        Error.stackTraceLimit = stackTraceLimit;

        // Capture the stack trace above function b
        Error.captureStackTrace(error, b); // Neither function c, nor b is included in the stack trace
        throw error;
      }


      a();

      ```
    remarks: ''
    isDeprecated: false
    syntax:
      content: >-
        static function captureStackTrace(targetObject: object, constructorOpt?:
        Function)
      parameters:
        - id: targetObject
          type: object
          description: ''
        - id: constructorOpt
          type: Function
          description: ''
    inheritanceDescription: <b>Inherited From</b> Error.captureStackTrace
  - name: prepareStackTrace(Error, CallSite[])
    uid: durable-functions.DurableError.prepareStackTrace
    package: durable-functions
    summary: |-


      See https://v8.dev/docs/stack-trace-api#customizing-stack-traces
    remarks: ''
    isDeprecated: false
    syntax:
      content: >-
        static function prepareStackTrace(err: Error, stackTraces: CallSite[]):
        any
      parameters:
        - id: err
          type: Error
          description: ''
        - id: stackTraces
          type: CallSite[]
          description: ''
      return:
        description: ''
        type: any
    inheritanceDescription: <b>Inherited From</b> Error.prepareStackTrace
extends: Error
